src/wp-includes/ms-files.php

```diff
@@ -18,7 +18,7 @@
 
 ms_file_constants();
 
-if ( '1' === $current_blog->archived || '1' === $current_blog->spam || '1' === $current_blog->deleted ) {
+if ( ( '1' === $current_blog->archived || '1' === $current_blog->spam || '1' === $current_blog->deleted ) && ! is_super_admin() ) {
```

Comment:
@Clorith suggested: do the super admin check first, as it saves having to do 3 more comparisons